### ISSUE TITLE
[v6] Fields don't update in response to `submitting` flag

### ISSUE THREAD

**Issue by clayne11** [erikras/redux-form#1122]
The `ConnectedField.shouldComponentUpdate` looks like this:

```javascript
shouldComponentUpdate(nextProps) {
  return !deepEqual(this.props, nextProps)
}
```

The problem is that the `submitting` flag gets set in `this.context._reduxForm`. This essentially means that Fields don't update in response to the submitting flag, and so if you want to disable / enable fields based on the submitting state, the Field will not currently update.

We need to add context to `shouldComponentUpdate` and check the `submitting` flag.

```javascript
shouldComponentUpdate(nextProps, nextState, nextContext) {
  return this.context._reduxForm.submitting !== nextContext._reduxForm.submitting ||
    !deepEqual(this.props, nextProps)
}
```

This same change also needs to be made in `ConnectedFieldArray`.

**Comment by clayne11** [comment]
The problem seems to go even deeper than this. I tried this change, but toggling the `submitting` flag doesn't even force an update on the `ConnectedField`. It takes cycling `submitting` on and off before the context in `ConnectedField` updates properly.

That's because `Field` has the same issue with its `shouldComponentUpdate`:

```javascript
// Field.js
shouldComponentUpdate(nextProps) {
  return shallowCompare(this, nextProps)
}
```

**Comment by clayne11** [comment]
This approach isn't working very well. Looking at it again, I think it makes more sense to just add the `submitting` flag to the `connect` function in `ConnectedField`.

**Comment by erikras** [comment]
> This approach isn't working very well. Looking at it again, I think it makes more sense to just add the `submitting` flag to the connect function in `ConnectedField`.

There's your answer. The other alternative is what I did today to get the field component to update when its sync error changes (pending your work, obviously) by looking at `context._reduxForm` in `Field` and passing it down to `ConnectedField` as a prop. But I think that if it's already in the Redux store, just `connect()` to it.

**Comment by erikras** [comment]
Fix published in [`v6.0.0-rc.5`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-rc.5).
